src/logic/calculate.js

```diff
@@ -0,0 +1,129 @@
+import operate from './operate';
+
+function isNumber(item) {
+  return !!item.match(/[0-9]+/);
+}
+
+/**
+ * Given a button name and a calculator data object, return an updated
+ * calculator data object.
+ *
+ * Calculator data object contains:
+ *   total:s      the running total
+ *   next:String       the next number to be operated on with the total
+ *   operation:String  +, -, etc.
+ */
+export default function calculate(obj, buttonName) {
+  if (buttonName === 'AC') {
+    return {
+      total: 0,
+      next: null,
+      operation: null,
```

Comment:
I changed it to 0 to show 0, otherwise, nothing would happen when the AC button is clicked. This happens on my end, I'm not sure if "null" should work, but for me it isn't. On top of that, there is no requirement to keep the files as they are, we should be allowed to change them

_"To help you concentrate on how to work with React events, we have developed the math logic you need to make the Calculator component work."_ -  Doesn't say we can't change them, the files are given just to help us concentrate on react and not JS.